### PR TITLE
feat(worker): Allow fetching config variables (e.g. in cloud) and imp…

### PR DESCRIPTION
**Pull request by berndruecker**
…roved expression capabilities (https://github.com/zeebe-io/zeebe-http-worker/issues/5)

### CODE REVIEW

**Comment by saig0** [comment]
I applied some small changes.

File: src/main/java/io/zeebe/http/EnvironmentVariableLoader.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  protected final HttpClient client = HttpClient.newHttpClient();
+  protected final ObjectMapper objectMapper = new ObjectMapper();
+  
+  private long cacheLoaded = 0;
+  private Map<String, String> cachedValues = null;
+  
+  /**
+   * You can set this to use this value instead of the environment variable, e.g. in testing
+   */
+  public static String MOCK_ENV_VARS_URL = null;
```
[the remark this answers is not in the document]
Comment (berndruecker) [reply src/main/java/io/zeebe/http/EnvironmentVariableLoader.java]: No - this is only used in Unit tests

File: src/main/java/io/zeebe/http/ConfigurationMaps.java [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+class ConfigurationMaps {
+  
+  public Map<String, String> customHeaders;
+  public Map<String, Object> variables;
+  public Map<String, String> environmentVariables;   
+  
+  /**
+   * Replace variables in 
+   **/
+  public String replaceVariables(String input) {
```
[the remark this answers is not in the document]
Comment (berndruecker) [reply src/main/java/io/zeebe/http/ConfigurationMaps.java]: I kind of agree but find that overkill for the moment. Actually I am waiting for the core broker to support EL everywhere, then the worker doesn't have to do all of that anymore

File: src/main/java/io/zeebe/http/HttpJobHandler.java [hunk 3]
```diff
@@ -58,14 +58,16 @@ public void handle(JobClient jobClient, ActivatedJob job)
...
   }
-
+  
   private HttpRequest buildRequest(ActivatedJob job) {
-    final Map<String, String> customHeaders = job.getCustomHeaders();
-    final Map<String, Object> variables = job.getVariablesAsMap();
+    ConfigurationMaps configurationMaps = new ConfigurationMaps();
+    configurationMaps.customHeaders = job.getCustomHeaders();
+    configurationMaps.variables = job.getVariablesAsMap();
+    configurationMaps.environmentVariables = environmentVariableLoader.loadVariables();
```
[the remark this answers is not in the document]
Comment (berndruecker) [reply src/main/java/io/zeebe/http/HttpJobHandler.java]: I know that this is kind of the typical way of doing it, but do not see any advantage if I basically use a DTO static class, so I skipped it

File: src/main/java/io/zeebe/http/ZeebeHttpWorker.java [hunk 4]
```diff
@@ -39,7 +39,12 @@ public void start() {
             .build();
 
     final HttpJobHandler jobHandler = new HttpJobHandler();
-    jobWorker = client.newWorker().jobType("http").handler(jobHandler).fetchVariables(HttpJobHandler.VARIABLE_NAMES).open();
+    jobWorker = client.newWorker() //
+        .jobType("http") //
+        .handler(jobHandler) //
+        //.fetchVariables(HttpJobHandler.VARIABLE_NAMES) // we need to fetch everything
```
[the remark this answers is not in the document]
Comment (berndruecker) [reply src/main/java/io/zeebe/http/ZeebeHttpWorker.java]: Yeah - but I would love to keep it as I think we will revisit that later when the broker learns expressions

**Pull request merged**